lib/rubber/configuration/cluster.rb

```diff
@@ -0,0 +1,195 @@
+require 'rubber/configuration/instance_item'
+
+module Rubber
+  module Configuration
+
+    # Contains the ec2 cluster configuration defined in instance.yml
+    #
+    class Cluster
+      attr_reader :configuration_storage, :artifacts
+      include Enumerable
+      include MonitorMixin
+
+      def initialize(configuration_storage, opts={})
+        super()
+
+        @configuration_storage = configuration_storage
+        @opts = opts
+
+        @items = {}
+        @artifacts = {'volumes' => {}, 'static_ips' => {}, 'vpc' => {}}
+
+        @filters = Rubber::Util::parse_aliases(ENV['FILTER'])
+        @filters, @filters_negated = @filters.partition {|f| f !~ /^-/ }
+        @filters_negated = @filters_negated.collect {|f| f[1..-1] }
+
+        @filter_roles = Rubber::Util::parse_aliases(ENV['FILTER_ROLES'])
+        @filter_roles, @filter_roles_negated = @filter_roles.partition {|f| f !~ /^-/ }
+        @filter_roles_negated = @filter_roles_negated.collect {|f| f[1..-1] }
+
+        load()
+      end
+
+      def load(configuration_storage=@configuration_storage)
+        case configuration_storage
+          when /file:(.*)/
+            location = $1
+            File.open(location, 'r') {|f| load_from_file(f) } if File.exist?(location)
```

Comment:
Should this raise an error if the file doesn't exist?

Reply:
Not necessarily.  What about the case when the cluster is first created and the instance file doesn't exist yet?